### ISSUE TITLE
Can not change ID of participant

### ISSUE THREAD

**Issue by nclaidiere** [open-cogsci/omm-server#154]
Hi, we had to change the RFID chips of some monkeys and we can not update it on the server.

At first I thought this was coming from a server connexion problem since we had similar problems when updating participants before, but strangely I can change all the parameters (alternate ID, name, etc) but not the main identifier, which suggest that there is something preventing only this parameter from updating (and nothing wrong with the server connexion). The error message is also a new one (see below).

I tried inactivating the participant to change it but that didn't work either.

I created a new participant, attributed it to a experiment and could still change the identifier.

I restarted the server, just in case, but that didn't work either. I also tried when no monkeys were performing the experiment.

Hum... any idea?

![ID_change_2024-02-14_07-19-22](https://github.com/open-cogsci/omm-server/assets/50901388/b70863f2-fa3f-4740-bc18-aa68a4405dd8)

**Comment by kristian-lange** [comment]
@smathot How do we want to handle the development process? So far I have a fix for this ticket on my own fork of OMM (https://github.com/open-cogsci/omm-server/commit/21abc52572e96af7ae8c69270871c762b650bdf5). Do you want PRs? Or should I just merge into to OMM' master branch?

**Comment by smathot** [comment]
I'm open to suggestions if you have a preference, but my suggestion is that you send me a PR and then I will review and merge.

**Comment by nclaidiere** [comment]
With the new version of OMM server the issue still persists, the alt. identifier can be changed but not the primary identifier. Here is a procedure to recreate the problem: create a participant, assign to an experiment, do a few trials, try to change the main id.

**Comment by kristian-lange** [comment]
Hi Nicolas, it seems to work with me. I made a screencast. Please have a look. Or did you mean a different identifier?

https://github.com/user-attachments/assets/6a8c6b7e-8251-4016-ae7c-a55ad96e6cab

**Comment by nclaidiere** [comment]
Hi, this is the identifier I am talking about. Is your participant part of an ongoing experiment? if not, try again once you've done a few trials of an experiment, it should not work.

**Comment by smathot** [comment]
Indeed, the primary identifier cannot be updated for participants who have completed trials, because the identifier is itself used as a primary key in the database and thus linked to other tables. Here's the error message:

```
update `participants` set `identifier` = 'test', `active` = true, `created_at` = '2025-02-12 15:42:09', `updated_at` = '2025-02-14 11:34:40', `meta` = 'null' where `id` = 97 - Cannot delete or update a parent row: a foreign key constraint fails (`omm`.`sessions`, CONSTRAINT `sessions_participant_id_foreign` FOREIGN KEY (`participant_id`) REFERENCES `participants` (`identifier`) ON DELETE CASCADE)
```

**Comment by kristian-lange** [comment]
I understand now. I added an 'ON UPDATE CASCADE' to the issue causing foreign key constraint. See PR https://github.com/open-cogsci/omm-server/pull/194.
